File: BaekJoon/Silver/10828/Python/10828.py
import sys

n = int(sys.stdin.readline())
stack = []
for i in range(0, n):
    str1 = sys.stdin.readline().rstrip()
    if str1[0:4] == 'push':
        n = int(str1.split().pop())
        stack.append(n)
    elif str1 == 'top':
        if len(stack) != 0:
            print(stack[-1])
        else:
            print(-1)
    elif str1 == 'size':
        print(len(stack))
    elif str1 == 'empty':
        print(1 if len(stack) == 0 else 0)
    else:
        if len(stack) != 0:
            print(stack.pop())
        else:
            print(-1)

File: BaekJoon/Silver/10828/C++/10828.cpp
// no.10828: 스택 (S4)

#include <cstdio>
#include <stack>
#include <cstring>
using namespace std;

int main() {
    int n;
    scanf("%d", &n);
    stack<int> stk;
    for(int i=0; i<n ;i++) {
        char inst[20];
        scanf("%s", inst);
        if(strcmp(inst, "push") == 0) {
            int val;
            scanf("%d", &val);
            stk.push(val);
        }
        if(strcmp(inst, "pop") == 0) {
            if(stk.empty()) {
                printf("-1\n");
                continue;
            }
            printf("%d\n", stk.top());
            stk.pop();
        }
        if(strcmp(inst, "size") == 0) {
            printf("%lu\n", stk.size());
        }
        if(strcmp(inst, "empty") == 0) {
            printf("%d\n", stk.empty() ? 1 : 0);
        }
        if(strcmp(inst, "top") == 0) {
            if(stk.empty()) {
                printf("-1\n");
                continue;
            }
            printf("%d\n", stk.top());
        }
    }
    return 0;
}